tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,574 @@
+inline TVec<T> &OperateInPlace(TVec<T> &v, F &&f)
+{
+   std::transform(v.begin(), v.end(), v.begin(), std::forward<F>(f));
+   return v;
+}
+
+} // End of VecOps NS
+
+} // End of Internal NS
+
+namespace Experimental {
+
+namespace VecOps {
+// clang-format off
+/*
+\class ROOT::Experimental::VecOps::TVec
+\ingroup vecops
+\brief A "std:::vector"-like collection of values implementing handy operation to analyse them
+\tparam T The type of the contained objects
+
+A TVec is a container designed to make analysis of values' collections fast and easy.
+Its storage is contiguous in memory and its interface is designed such to resemble to the one
+of the stl vector. In addition the interface features methods and external functions to ease
+the manipulation and analysis of the data in the TVec.
+
+For example, suppose to have in hands an event featuring eight muons with a certain pseudorapidity,
+momentum and charge. Suppose you want to extract the transverse momenta of the muons satisfying certain
+criteria. Something which would require, among the other things, the management of an explicit loop,
+becomes straightforward with TVec:
+~~~{.cpp}
+TVec<short> mu_charge {1, 1, -1, -1, -1, 1, 1, -1};
+TVec<float> mu_pt {56, 45, 32, 24, 12, 8, 7, 6.2};
+TVec<float> mu_eta {3.1, -.2, -1.1, 1, 4.1, 1.6, 2.4, -.5};
+auto goodMuons_pt = mu_pt[ (mu_pt > 10.f && abs(mu_eta) <= 2.f && mu_charge == -1)
+~~~
+Now the clean collection of transverse momenta can be used within the rest of the data analysis, for
+example to fill a histogram.
+
+## Owning and adopting memory
+TVec has contiguous memory associated to it. It can own it or simply adopt it. In the latter case,
+it can be constructed with the address of the memory associated to it and its lenght. For example:
+~~~{.cpp}
+std::vector<int> myStlVec {1,2,3};
+TVec<int> myTVec(myStlVec.data(), myStlVec.size());
+~~~
+In this case, the memory associated to myStlVec and myTVec is the same, myTVec simply "adopted it".
+If any method which implies a re-allocation is called, e.g. *emplace_back* or *resize*, the adopted
+memory is released and new one is allocated. The previous content is copied in the new memory and
+preserved.
+*/
+// clang-format on
+template <typename T>
+class TVec {
+public:
+   using Impl_t = typename std::vector<T, ROOT::Detail::VecOps::TAdoptAllocator<T>>;
+   using value_type = typename Impl_t::value_type;
+   using size_type = typename Impl_t::size_type;
+   using difference_type = typename Impl_t::difference_type;
+   using reference = typename Impl_t::reference;
+   using const_reference = typename Impl_t::const_reference;
+   using pointer = typename Impl_t::pointer;
+   using const_pointer = typename Impl_t::const_pointer;
+   using iterator = typename Impl_t::iterator;
+   using const_iterator = typename Impl_t::const_iterator;
+   using reverse_iterator = typename Impl_t::reverse_iterator;
+
+private:
+   Impl_t fData;
+
+public:
+   // ctors
+   TVec() {}
```

Comment:
It'd be *really* good to have at least a move ctor and maybe even a copy ctor.
Same with assignment.

Reply:
You are right. And @bluehood  also note this. The change I had got lost in a rebase: this will be of course addressed. Sorry about that!